### PR TITLE
Add require_password_in_authentication_commands driver flag to allow skipping password check in authentication commands

### PR DESCRIPTION
**Pull request by Krock21**
Issue: #927

### CODE REVIEW

**Review by l0kix2 — APPROVED** [review]

**Review by Kontakter — APPROVED** [review]
To run tests in internal CI

**Comment by Krock21** [comment]
Please cherrypick it to (23.2), 24.1 and 24.2, if possible

File: yt/yt/client/driver/etc_commands.cpp [hunk 1]
```diff
@@ -96,6 +96,9 @@ void TGetSupportedFeaturesCommand::DoExecute(ICommandContextPtr context)
     for (auto staticFeature : StaticFeatures) {
         features->AddChild(TString(staticFeature.first), BuildYsonNodeFluently().Value(staticFeature.second));
     }
+    features->AddChild(
```
Comment (gritukan) [inline yt/yt/client/driver/etc_commands.cpp:99]: Is it for UI?

**Review by gritukan — APPROVED** [review]
LGTM

File: yt/yt/client/driver/etc_commands.cpp (hunk 1 above)
Comment (Krock21) [reply yt/yt/client/driver/etc_commands.cpp:99]: yes

**Comment by Kontakter** [comment]
There are multiple errors:
```
yt/yt/tests/integration/proxies/test_cypress_token_auth.py:40: [E302] expected 2 blank lines, found 1
yt/yt/tests/integration/proxies/test_cypress_token_auth.py:92: [F541] f-string is missing placeholders
yt/yt/tests/integration/proxies/test_cypress_token_auth.py:94: [F541] f-string is missing placeholders
yt/yt/tests/integration/proxies/test_cypress_token_auth.py:104: [E303] too many blank lines (3)
yt/yt/tests/integration/proxies/test_cypress_token_auth.py:137: [F541] f-string is missing placeholders
yt/yt/tests/integration/proxies/test_cypress_token_auth.py:139: [F541] f-string is missing placeholders
```

```
contrib/python/requests/py3/requests/models.py:434: in prepare_url
    scheme, auth, host, port, path, query, fragment = parse_url(url)
contrib/python/urllib3/py3/urllib3/util/url.py:397: in parse_url
    return six.raise_from(LocationParseError(source_url), None)
E   urllib3.exceptions.LocationParseError: Failed to parse: http://localhost:24268ytct-69db-360354809c55a39f22100f678f77b547

During handling of the above exception, another exception occurred:
yt/yt/tests/integration/proxies/test_cypress_token_auth.py:73: in test_revoke_token
    self._check_allow(t)  # Activate cache.
yt/yt/tests/integration/proxies/test_cypress_token_auth.py:30: in _check_allow
    rsp = self._make_request(path, token)
yt/yt/tests/integration/proxies/test_cypress_token_auth.py:25: in _make_request
    return requests.get(
contrib/python/requests/py3/requests/api.py:73: in get
    return request("get", url, params=params, **kwargs)
contrib/python/requests/py3/requests/api.py:59: in request
    return session.request(method=method, url=url, **kwargs)
contrib/python/requests/py3/requests/sessions.py:575: in request
    prep = self.prepare_request(req)
contrib/python/requests/py3/requests/sessions.py:486: in prepare_request
    p.prepare(
contrib/python/requests/py3/requests/models.py:368: in prepare
    self.prepare_url(url, params)
contrib/python/requests/py3/requests/models.py:436: in prepare_url
    raise InvalidURL(*e.args)
E   requests.exceptions.InvalidURL: Failed to parse: http://localhost:24268ytct-69db-360354809c55a39f22100f678f77b547
```

```
yt/yt/tests/integration/proxies/test_cypress_token_auth.py:58: in test_simple
    assert get("//sys/cypress_tokens/@count") == 1
E   AssertionError: assert 4 == 1
E    +  where 4 = get('//sys/cypress_tokens/@count')
```

**Comment by Krock21** [comment]
Fixed failing tests

**Comment by Kontakter** [comment]
Shipped!

**Comment by Krock21** [comment]
> Test `test_cypress_token_auth.py::TestCypressTokenAuth test_simple (gritukan)` fails:
> 
> ```
> yt/yt/tests/integration/proxies/test_cypress_token_auth.py:59: in test_simple
>     assert get("//sys/cypress_tokens/@count") == 1
> E   AssertionError: assert 3 == 1
> E    +  where 3 = get('//sys/cypress_tokens/@count')
> ```

Strange. I don't get the same error locally

It is probably because tests in the same `class` reuse the same cluster state, so they should clean up after themselves

`test_simple` adds a token to user `u` and checks tokens count, expecting 1
`test_authentication_commands` adds 2 tokens and finishes.

And if `test_authentication_commands` was executed first, `test_simple` fails with 3 tokens instead of 1

I changed tests to use different users for their checks

**Comment by savnadya** [comment]
Shipped!

**Comment by savnadya** [comment]
Shipped!

**Comment by l0kix2** [comment]
Are tests ok now?

**Pull request closed without merging**